src/platform/imx8/lib/dma.c

```diff
@@ -29,9 +29,9 @@ struct dma dma[PLATFORM_NUM_DMACS] = {
 {
 	.plat_data = {
 		.id		= DMA_ID_HOST,
-		.dir		= DMA_DIR_MEM_TO_MEM,
+		.dir		= DMA_DIR_HMEM_TO_LMEM | DMA_DIR_LMEM_TO_HMEM,
 		.devs		= DMA_DEV_HOST,
-		.channels	= 32,
+		.channels	= 16,
 	},
 	.ops	= &dummy_dma_ops,
```

Comment:
Please create two separate patches, one changing the number of channels and the other updating the dir.

Reply:
The directions I still change in the main patch. Separated change of channel numbers since that doesn't affect functionality but only memory efficiency.